test/test_io_common.py

```diff
@@ -0,0 +1,212 @@
+"""
+Common IO tests that apply to all grid formats.
+
+This module tests functionality that should work across all supported formats:
+- Basic read/write operations
+- Format conversions and round-trips
+- UGRID compliance
+- Common error handling
+"""
+
+import pytest
+import numpy as np
+import xarray as xr
+import uxarray as ux
+from pathlib import Path
+import tempfile
+import os
+from numpy.testing import assert_array_equal, assert_allclose
+from uxarray.constants import ERROR_TOLERANCE, INT_DTYPE, INT_FILL_VALUE
+
+try:
+    import constants
+except ImportError:
+    from . import constants
+
+current_path = Path(os.path.dirname(os.path.realpath(__file__)))
+
+# Define all testable format combinations
+# Format: (format_type, subpath, filename)
+IO_READ_TEST_FORMATS = [
+    ("ugrid", "ugrid/quad-hexagon", "grid.nc"),
+    ("ugrid", "ugrid/outCSne30", "outCSne30.ug"),
+    ("ugrid", "ugrid/outRLL1deg", "outRLL1deg.ug"),
+    ("mpas", "mpas/QU/480", "grid.nc"),
+    ("esmf", "esmf/ne30", "ne30pg3.grid.nc"),
+    ("exodus", "exodus/outCSne8", "outCSne8.g"),
+    ("exodus", "exodus/mixed", "mixed.exo"),
+    ("scrip", "scrip/outCSne8", "outCSne8.nc"),
+    ("icon", "icon/R02B04", "icon_grid_0010_R02B04_G.nc"),
+    ("fesom", "fesom/pi", None),  # Special case - multiple files
+]
```

Comment:
If we plan to only cover grid functionality here, this is fine, otherwise we need to math the corresponding data file too for more comprehensive dataset tests.

Reply:
grid only is what I planned.